### PR TITLE
dhcpcd: make privsep a build option

### PR DESCRIPTION
**Pull request by CameronNemo**
Still forks away from the supervisor :/
But there is no huge process tree, just a launcher and master.

### CODE REVIEW

File: srcpkgs/dhcpcd/template [hunk 1]
```diff
@@ -15,9 +15,15 @@ distfiles="https://roy.marples.name/downloads/dhcpcd/dhcpcd-${version}.tar.xz"
...
-system_accounts="_dhcpcd"
-_dhcpcd_homedir="/var/db/dhcpcd"
+
+build_options="privsep"
+build_options_default="privsep"
+
+if [ "$build_option_privsep" ]; then
+	configure_args+=" --privsepuser=_dhcpcd"
+	system_accounts="_dhcpcd"
+	_dhcpcd_homedir="/var/db/dhcpcd"
```
Comment (sgn) [inline srcpkgs/dhcpcd/template:25]: I prefer to keep them for no reason.

File: srcpkgs/dhcpcd/template [hunk 2]
```diff
@@ -15,10 +15,17 @@ distfiles="https://roy.marples.name/downloads/dhcpcd/dhcpcd-${version}.tar.xz"
...
-# privsep
+
 system_accounts="_dhcpcd"
 _dhcpcd_homedir="/var/db/dhcpcd"
 
+build_options="privsep"
+build_options_default="privsep"
+
+if [ "$build_option_privsep" ]; then
+	configure_args+=" --privsepuser=_dhcpcd"
```
Comment (Piraty) [inline srcpkgs/dhcpcd/template:26]: `vopt_if`?

**Review by Piraty — CHANGES_REQUESTED** [review]

File: srcpkgs/dhcpcd/template (hunk 1 above)
Comment (CameronNemo) [reply srcpkgs/dhcpcd/template]: kept

File: srcpkgs/dhcpcd/template (hunk 2 above)
Comment (CameronNemo) [reply srcpkgs/dhcpcd/template]: used vopt_if

**Comment by the-maldridge** [comment]
This doesn't really solve the original problem though, which is it forking away from the supervisor.   I'm not sure why you'd add this package since it doesn't change the status of the bug that needs to be fixed.

**Pull request closed without merging**